ipa-core/src/protocol/ipa_prf/malicious_security/lagrange.rs

```diff
@@ -176,6 +160,29 @@ where
     }
 }
 
+/// Computes the dot product of two arrays of the same size.
+/// It is isolated from Lagrange because there could be potential SIMD optimizations used
+fn dot_product<F: PrimeField, const N: usize>(a: &[F; N], b: &[F; N]) -> F {
+    // Staying in integers allows rustc to optimize this code properly
+    // with any reasonable N, we won't run into overflow with dot product.
+    // (N can be as large as 2^32 and still no chance of overflow for 61 bit prime fields)
+    debug_assert!(
+        F::PRIME.into() < (1 << 64),
+        "The prime {} is too large for this dot product implementation",
+        F::PRIME.into()
```

Comment:
We are multiplying two field values and then summing. I'd say that the conservative requirement is `2 * F::BITS + N.next_power_of_two().ilog2() <= u128::BITS`.

Reply:
that's right, I think this assertion just assumes `N` being reasonable and the second argument of this expression negligible compared to `F::BITS`. I don't mind changing it, the reasoning behind this was just to optimize for readers